pkg/frontend/frontend.go

```diff
@@ -231,8 +231,14 @@ func (f *frontend) chiAuthenticatedRoutes(router chi.Router) {
 				r.With(f.apiVersionMiddleware.ValidateAPIVersion).Route("/", func(r chi.Router) {
 					r.Delete("/", f.deleteOpenShiftCluster)
 					r.Get("/", f.getOpenShiftCluster)
-					r.Patch("/", f.putOrPatchOpenShiftCluster)
-					r.Put("/", f.putOrPatchOpenShiftCluster)
+
+					if f.env.IsLocalDevelopmentMode() {
+						r.With(middleware.MockMSIMiddleware).Patch("/", f.putOrPatchOpenShiftCluster)
```

Comment:
What is the idea here to use the middleware for a Patch request? Since we are assuming that the Cluster MSI/Identity object will not be changed. 
As per MSI RP it is considered successful no operation, so it should be fine to use the middleware here.
But can we just skip this middleware part for Patch, as because not getting a identityURL for an empty Identity object is also a successful no operation and valid here?

Reply:
I also want to raise an additional point here around patch request, that might be related here.
How to handle the case where cx patches the identity object such that it removes the msi from the object(doesn't delete the identity though)(This request will return from ARO RP as Identity object changed which is immutable), but cx later on patches with exact same identity object as was present previously?

The old MSI Certificate will no longer work as it will be deleted when cx patched the identity object first time. Should we somehow refresh credential via a patch request or just ignore the whole scenario as an unsupported cluster patching?

Reply:
My understanding is that the Cluster MSI headers are still returned on a PATCH, not just a CREATE, so we want to mock both instead of just create. I'm not sure on your second point, though...

Reply:
1. The headers are part of PATCH operation: https://eng.ms/docs/products/arm/rbac/managed_identities/msionboardingsteps
2. Based on our previous architecture discussions, we agreed that we shouldn't support removing/modifying the cluster MSI. If we want to support that in the future, we can.

Reply:
>I also want to raise an additional point here around patch request, that might be related here.
How to handle the case where cx patches the identity object such that it removes the msi from the object(doesn't delete the identity though)(This request will return an error from ARO RP as Identity object got changed, which is immutable), but cx later on patches with exact same identity object as was present previously?
>
>The old MSI Certificate will no longer work as it will be deleted when cx patched the identity object first time. Should we somehow refresh credential via the second patch request or just ignore the whole scenario as an unsupported cluster patching?

I think this is currently impossible since the `Identity` is immutable, so we can just ignore the whole scenario unless we later decide to support removing/modifying the MSI.

Reply:
Yes we can mark the cluster unsupported if the MSI is tried to be removed from the cluster object as part of PUT request.

@kimorris27, yes the flow will fail as part of ARO-RP but the MSI-RP would consider it deleted and then there will be a inconsistency between MIRP and ARO-RP. Apologies, IDK why I didn't attach the ref earlier:- https://msazure.visualstudio.com/One/_git/AzureUX-ARMWiki?path=/src/rbac/managed_identities/MSIOnboardingUserAssigned.md&version=GBmaster&line=1&_a=preview&anchor=client-error-when-more-than-1-identity-is-configured-on-the-resource